source/slang/slang-check-decl.cpp

```diff
@@ -8475,19 +8829,327 @@ namespace Slang
             }
             if (seqStmtChild->stmts.getCount() == 0)
                 continue;
-            seqStmt->stmts.insert(insertOffset, seqStmtChild);
+
+            auto seqStmt = _ensureCtorBodyIsSeqStmt(m_astBuilder, ctor);
+            seqStmt->stmts.insert(ctorInfo.m_insertOffset++, seqStmtChild);
         }
 
-        if (structDeclInfo.defaultCtor)
+        // Note: we assume only 1 inheritance decl currently.
```

Comment:
There is a lot of code added to this checkAggType method. We really need to split them into separate functions for readability.

Reply:
Create sub-task #5135 to address comment.